final/model/dataset.py

```diff
@@ -138,21 +194,58 @@ def get_spectrogram(self, audio, label):
         return spectrogram, label
 
 
-    def get_log_mel_spectrogram(self, spectrogram, label):
+    def get_log_mel_spectrogram(
+            self,
+            spectrogram: tf.Tensor,
+            label: tf.Tensor
+            ) -> tuple[tf.Tensor, tf.Tensor]:
+        """
+        IN:
+            - the spectrogram of the signal
+            - the label reresented as a 1-D tensor
+        OUT:
+            - the log mel spectrogram of the signal
+            - the label
+        """
         mel_spectrogram = spectrogram @ self.linear_to_mel_weight_matrix
         log_mel_spectrogram = tf.math.log(mel_spectrogram + 1e-6)
         return log_mel_spectrogram, label
 
 
-    def get_mfccs(self, log_mel_spectrogram, label):
+    def get_mfccs(
+            self,
+            log_mel_spectrogram: tf.Tensor,
+            label: tf.Tensor
+            ) -> tuple[tf.Tensor, tf.Tensor]:
+        """
+        IN:
+            - the log mel spectrogram of the signal
+            - the label reresented as a 1-D tensor
+        OUT:
+            - the mfccs representation of the signal
+            - the label
+        """
         mfccs = tf.signal.mfccs_from_log_mel_spectrograms(log_mel_spectrogram)
         return mfccs[:, :self.num_coefficients], label
 
 
-    def random_crop(self, audio, label):
-        if self.random_crop_length > 0:
+    def random_crop(
```

Comment:
We do not need this function anymore as data augmentation is done with sliding windows.

if you combine both techniques the model would be trained on a slice of the window,
which can be simply achieved by reducing the window size in the new code